TypeSystem/variant.h

```diff
@@ -50,6 +54,16 @@ SOFTWARE.
 
 namespace current {
 
+namespace variant {
+#ifdef VARIANT_CHECKS_AT_RUNTIME_INSTEAD_OF_COMPILE_TIME
+// When variant inner type checks are run-time, no nested variants are allowed.
+using InnerType = CurrentStruct;
```

Comment:
If you prefer camel-case for this reason, what about `InnerBaseType` or `InnerSuperType`? I'd like to specify explicitly that it's not the inner type of the contained object, but the base type.